File: client/src/main/java/au/com/codeka/warworlds/client/ui/ShowInfo.java
package au.com.codeka.warworlds.client.ui;

import android.view.View;

import androidx.annotation.Nullable;

public class ShowInfo {
  private final View view;
  private final boolean toolbarVisible;

  private ShowInfo(Builder builder) {
    this.view = builder.view;
    this.toolbarVisible = builder.toolbarVisible;
  }

  public static Builder builder() {
    return new Builder();
  }

  @Nullable
  public View getView() {
    return view;
  }

  public boolean getToolbarVisible() {
    return toolbarVisible;
  }

  public static class Builder {
    private View view;
    private boolean toolbarVisible;

    private Builder() {
      toolbarVisible = true;
    }

    public Builder view(View view) {
      this.view = view;
      return this;
    }

    public Builder toolbarVisible(boolean toolbarVisible) {
      this.toolbarVisible = toolbarVisible;
      return this;
    }

    public ShowInfo build() {
      return new ShowInfo(this);
    }
  }
}

File: client/src/main/java/au/com/codeka/warworlds/client/ui/ShowInfo.kt
package au.com.codeka.warworlds.client.ui

import android.view.View

class ShowInfo private constructor(builder: Builder) {
  val view: View?
  val toolbarVisible: Boolean

  class Builder {
    var view: View? = null
    var toolbarVisible = true
    fun view(view: View?): Builder {
      this.view = view
      return this
    }

    fun toolbarVisible(toolbarVisible: Boolean): Builder {
      this.toolbarVisible = toolbarVisible
      return this
    }

    fun build(): ShowInfo {
      return ShowInfo(this)
    }
  }

  companion object {
    @JvmStatic
    fun builder(): Builder {
      return Builder()
    }
  }

  init {
    view = builder.view
    toolbarVisible = builder.toolbarVisible
  }
}